vnext/Microsoft.ReactNative/Utils/ManagedResource.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+template<typename T, typename closeFn_t>
```

Comment:
Why this vs a unique_ptr with custom deleter?

Reply:
@NickGerleman great question - it took me a few tries and it's because `unique_ptr`  _requires_ the held value to be a pointer type. 
in other words:

```cpp
  std::unique_ptr<HMODULE, decltype(&FreeLibrary)> appModel(
      LoadLibraryExW(L"Api-ms-win-appmodel-runtime-l1-1-2.dll", 0, LOAD_LIBRARY_SEARCH_SYSTEM32), FreeLibrary);

  decltype(&AppPolicyGetWindowingModel) pfnAppPolicyGetWindowingModel =
      reinterpret_cast<decltype(&AppPolicyGetWindowingModel)>(GetProcAddress(appModel.get(), "AppPolicyGetWindowingModel"));
```

this fails to compile for two reasons:
1- the ctor to unique_ptr<T, D> holds a T*. In this case T=HMODULE and LoadLibraryEx returns a T, not a T*. And we cannot take the address of a temporary.
2- unique_ptr's .get() returns a HMODULE*

So unless I'm missing something obvious (it is late after all), I think unique_ptr is really supposed to be used only for real pointer types, not handles